js/runner.go

```diff
@@ -757,7 +760,7 @@ func (u *ActiveVU) RunOnce() error {
 }
 
 func (u *VU) runFn(
-	ctx context.Context, isDefault bool, fn goja.Callable, args ...goja.Value,
+	ctx context.Context, isDefault bool, fn goja.Callable, cancel func(), args ...goja.Value,
```

Comment:
The alternative to this was to add one more field to VU but we already provide the context here ... so arguably this whole function needs some changes. But again - refactoring the whole package to get this in is likely going ot be way too complicated, take even longer and I would expect it will break something especially around common.Bind and co.

Reply:
[f3e6833](https://github.com/grafana/k6/pull/2228/commits/f3e6833d6862db8913c1dd17ffd65a0631f4e737) ?